### ISSUE TITLE
Dates and ISO 8601

### ISSUE THREAD

**Issue by Boris-c** [facebook/hermes#82]
ISO 8601 states:
YYYY-MM-DDThh:mm:ss.s

where:
ss   = two digits of second (00 through 59)
s    = one or more digits representing a decimal fraction of a second

But in Hermes, `new Date('2019-08-07T00:00:00.760738998Z')` gives `2019-08-15T19:18:58.998Z` because 760738998 is interpreted as 760738.998 seconds.

**Comment by Boris-c** [comment]
Just for clarity, a simpler example of the parsing bug:

`new Date('2019-08-07T00:00:00.5Z')` gives `2019-08-07T00:00:00.005Z`

5 milliseconds is wrong, should be 500 milliseconds

**Comment by avp** [comment]
Thanks for the bug report. We'll work on fixing this.

**Comment by Boris-c** [comment]
Just in case someone would like to find a JavaScript workaround for the bug, here is mine (I don't pretend it's universal, but it can be a starting point for others to wait for your fix instead of simply avoiding to use Hermes).

Just call that once at app startup:

```
if (__DEV__ || (new Date('2019-08-07T00:00:00.5Z')).getMilliseconds() === 5) {
  const millisDate = /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2}).(\d+)Z$/
  const _Date = Date;

  function MockDate(y, m, d, h, M, s, ms) {
    let date = undefined;

    switch (arguments.length) {
      case 0: return new _Date();
      case 1:
        if (typeof y === 'string') {
          const matches = y.match(millisDate);
          if (matches) {
            const [_, yy, mm, dd, hh, MM, ss, mmss] = matches;
            const fixed = Math.floor(new Number('.' + mmss) * 1000);
            const fixedStr = `${fixed}`.padStart(3, '0');
            return new _Date(`${yy}-${mm}-${dd}T${hh}:${MM}:${ss}.${fixed}Z`);
          }
        }
        return new _Date(y);
      default: return new _Date(y, m, d, h, M, s, ms);
    }
  }

  Object.setPrototypeOf(MockDate, _Date);
  Date = MockDate;
}
```

**Comment by avp** [comment]
Fixed in a084c4d27e311e8a458ed89fdf05587004db6450
